cli/src/ui.rs

```diff
@@ -258,12 +257,32 @@ pub enum PaginationChoice {
     Auto,
 }
 
+enum PagerConfig {
+    Disabled,
+    Builtin,
+    External(CommandNameAndArgs),
+}
+
+impl PagerConfig {
+    fn from_config(config: &StackedConfig) -> Result<PagerConfig, ConfigGetError> {
+        let PaginationChoice::Auto = config.get("ui.paginate")? else {
+            return Ok(PagerConfig::Disabled);
+        };
+        let pager_cmd: CommandNameAndArgs = config.get("ui.pager")?;
+        match pager_cmd {
```

Comment:
nit: this can be inlined as `match config.get("ui.pager")? { ... pager_cmd => Ok(..) }`